stonesoup/sensormanager/base.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+from abc import abstractmethod, ABC
+from typing import Callable, Set
+import random
+import numpy as np
+import itertools as it
+
+from ..base import Base, Property
+from ..sensor.sensor import Sensor
+
+
+class SensorManager(Base, ABC):
+    """The sensor manager base class.
+
+    The purpose of a sensor manager is to return a mapping of sensors and sensor actions
+    appropriate to a specific
+    scenario and with a particular objective, or objectives, in mind. This involves using
+    estimates of the situation and knowledge of the sensor system to calculate metrics associated
+    with actions, and then determine optimal, or near optimal, actions to take.
+
+    There is considerable freedom in both the theory and practice of sensor management and these
+    classes do not enforce a particular solution. A sensor manager may be 'centralised' in that
+    it controls the actions of multiple sensors, or individual sensors may have their own managers
+    which communicate with other sensor managers in a networked fashion.
+
+    """
+    sensors: Set[Sensor] = Property(doc="The sensor(s) which the sensor manager is managing. "
+                                        "These must be capable of returning available actions.")
+
+    reward_function: Callable = Property(
```

Comment:
I think this needs better definition so it's clear what form this callable should be in. May be worth adding a definition of RewardFunction type. Something like:

```python
RewardFunction = Callable[
    [Mapping[Sensor, Sequence[Action]], Set[Track], datetime.datetime],
    float
    ]
```

Reply:
We do have a RewardFunction base class to go with the UncertaintyRewardFunction which defines reward functions in a bit more detail - could we use this here?

Reply:
Could do, but would make it less clear that you could just pass any callable/function.

Maybe we could make the `RewardFunction` class a Callable, and then define it with the above. That would make it cover both use cases.